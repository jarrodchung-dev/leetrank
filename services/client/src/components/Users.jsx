import React from "react";

const Users = (props) => {
  return (
    <>
      <h3 className="title is-3 has-text-centered">Active Users</h3>
      <br />
      {props.users.map((user) => {
        return (
          <div className="level" key={user.id}>
            <div className="level-left">
              <div className="level-item">
                <p className="username">{user.username}</p>
              </div>
            </div>
          </div>
        );
      })}
    </>
  );
};

export default Users;
