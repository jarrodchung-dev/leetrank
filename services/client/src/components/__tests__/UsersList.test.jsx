import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Users from "../Users.jsx";

const users = [
  {
    id: 99,
    username: "test_user",
    email: "test_user@example.com",
    active: true
  },
  {
    id: 100,
    username: "test_user2",
    email: "test_user2@example.com",
    active: true
  }
];

describe("<UsersList />", () => {
  it("displays user data receievd as props", () => {
    const wrapper = shallow(<Users users={users} />);
    const levelItem = wrapper.find("div.level-item");
    expect(levelItem.length).toBe(2);
  });

  it("renders a snapshot successfully", () => {
    const tree = renderer.create(<Users users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
