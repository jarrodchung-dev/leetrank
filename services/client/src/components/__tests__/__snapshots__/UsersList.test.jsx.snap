// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersList /> renders a snapshot successfully 1`] = `
Array [
  <div
    className="level"
  >
    <div
      className="level-left"
    >
      <div
        className="level-item"
      >
        <p
          className="username"
        >
          test_user
        </p>
      </div>
    </div>
  </div>,
  <div
    className="level"
  >
    <div
      className="level-left"
    >
      <div
        className="level-item"
      >
        <p
          className="username"
        >
          test_user2
        </p>
      </div>
    </div>
  </div>,
]
`;
